### ISSUE TITLE
Model is overfitting heavily

### ISSUE THREAD

**Issue by wilczuuuuus** [wilczuuuuus/TissueMNISTCNN#1]
Why?

- increasing gap between training and validation accuracy
- Training accuracy continuously improves while validation accuracy oscillates
- Validation loss shows instability and large spikes
What to try to mitigate?

- [ ]  Implementing early stopping to prevent overfitting
- [ ]  Implementing data augmentation
- [ ] Adding regularization (increase dropout or add L2 regularization)
- [ ] Weighted Loss Function 
- [ ] Sampling Strategies - Implement either oversampling or undersampling
